### PR TITLE
[OPPRO-309] Change spark3.3 version to 3.3.1

### PR DESCRIPTION
**Pull request by jinchengchenghh**

### CODE REVIEW

File: backends-velox/src/test/scala/io/glutenproject/execution/VeloxWholeStageTransformerSuite.scala [hunk 1]
```diff
@@ -114,7 +114,11 @@ class VeloxWholeStageTransformerSuite extends WholeStageTransformerSuite {
       val wholeStages = plan.collect {
         case wst: WholeStageTransformerExec => wst
       }
-      assert(wholeStages.length == 1)
+      if (SparkShimLoader.getSparkVersion.startsWith("3.3.")) {
+        assert(wholeStages.length == 3)
```
Comment (PHILO-HE) [inline backends-velox/src/test/scala/io/glutenproject/execution/VeloxWholeStageTransformerSuite.scala:118]: Can we use the below code instead?
```
if (VersionUtils.majorVersion(SparkShimLoader.getSparkVersion) == 3 && VersionUtils.minorVersion(SparkShimLoader.getSparkVersion) == 2) {
 assert(wholeStages.length == 1)
} else {
 assert(wholeStages.length == 3)
}
```
Thus, we can avoid to change the condition after we upgrading to higher spark version, e.g., future 3.4.x, assuming it is still applicable to such higher versions.

File: .github/workflows/unittests.yml [hunk 2]
```diff
@@ -86,7 +86,7 @@ jobs:
       - name: Checkout gluten-it code in docker
         run: |
           docker exec velox-backend-test-$GITHUB_RUN_ID bash -c 'cd /opt/gluten && \
-          git clone -b main https://github.com/zhztheplayer/gluten-it.git gluten-it'
+          git clone -b main https://github.com/jinchengchenghh/gluten-it.git gluten-it'
       - name: TPC-H SF1.0 && TPC-DS SF0.1 Parquet local spark3.2
```
Comment (FelixYBW) [inline .github/workflows/unittests.yml:90]: Can we move it to OAP?

File: ep/build-velox/src/build_velox.sh [hunk 3]
```diff
@@ -9,8 +9,8 @@ COMPILE_VELOX=OFF
 ENABLE_EP_CACHE=OFF
 VELOX_HOME=/root/velox
 
-VELOX_REPO=https://github.com/oap-project/velox.git
-VELOX_BRANCH=main
+VELOX_REPO=https://github.com/jinchengchenghh/velox.git
+VELOX_BRANCH=ifthen
```
Comment (FelixYBW) [inline ep/build-velox/src/build_velox.sh:13]: set back to oap-project/velox

**Review by FelixYBW — APPROVED** [review]

File: ep/build-velox/src/build_velox.sh (hunk 3 above)
Comment (jinchengchenghh) [reply ep/build-velox/src/build_velox.sh:13]: wait for this https://github.com/oap-project/velox/pull/71

Comment (jinchengchenghh) [reply ep/build-velox/src/build_velox.sh]: switch to https://github.com/oap-project/velox/pull/74

**Review by FelixYBW — APPROVED** [review]

**Pull request merged**